crates/holochain/tests/tests/schedule.rs

```diff
@@ -0,0 +1,286 @@
+use hdk::prelude::{BoxApi, ExternIO, InlineZomeResult};
+use holochain::sweettest::{SweetCell, SweetConductor, SweetDnaFile};
+use holochain::test_utils::host_fn_caller::HostFnCaller;
+use holochain_sqlite::error::DatabaseError;
+use holochain_state::schedule::fn_is_scheduled;
+use holochain_types::inline_zome::InlineZomeSet;
+use holochain_types::signal::Signal;
+use holochain_zome_types::prelude::Schedule::Persisted;
+use holochain_zome_types::prelude::{Schedule, ScheduledFn};
+use holochain_zome_types::signal::AppSignal;
+use tokio::sync::broadcast;
+use tokio::time::error::Elapsed;
+
+/// Test schedule ephemeral fn
+/// Assuming a schedular interval of 100ms
+#[tokio::test(flavor = "multi_thread")]
+async fn schedule_ephemeral() {
+    holochain_trace::test_run();
+
+    // Start with a duration of 3ms and decrease by 1ms each time it's called.
+    let zome = create_schedule_zome(|api, input| {
+        let _ = api.emit_signal(AppSignal::new(ExternIO::encode(input.clone()).unwrap()));
+        let ms: u64 = match input {
+            Some(Schedule::Ephemeral(duration)) => duration.as_millis() as u64,
+            None => 3,
```

Comment:
```suggestion
            None => 3,
            Some(Schedule::Ephemeral(duration)) => duration.as_millis() as u64,
```

Possibly easier to follow? The `None` is expected first, then we take the input value on subsequent schedules calls.